static/app/views/issueDetails/streamline/eventGraph.tsx

```diff
@@ -68,348 +94,434 @@ function createSeriesAndCount(stats: EventsStats) {
 
-  const {series: unfilteredEventSeries} = useMemo(() => {
-    if (!unfilteredGroupStats?.['count()']) {
-      return {series: []};
-    }
+    // Ensure the dropdown can access the new filtered event count
+    useEffect(() => {
+      dispatch({type: 'UPDATE_EVENT_COUNT', count: eventCount});
+    }, [eventCount, dispatch]);
 
-    return createSeriesAndCount(unfilteredGroupStats['count()']);
-  }, [unfilteredGroupStats]);
-  const {series: unfilteredUserSeries} = useMemo(() => {
-    if (!unfilteredGroupStats?.['count_unique(user)']) {
-      return {series: []};
-    }
-    return createSeriesAndCount(unfilteredGroupStats['count_unique(user)']);
-  }, [unfilteredGroupStats]);
-  const userSeries = useMemo(() => {
-    if (!groupStats['count_unique(user)']) {
-      return [];
-    }
+    const {series: unfilteredEventSeries} = useMemo(() => {
+      if (!unfilteredGroupStats?.['count()']) {
+        return {series: []};
+      }
 
-    return createSeriesAndCount(groupStats['count_unique(user)']).series;
-  }, [groupStats]);
-
-  const chartZoomProps = useChartZoom({
-    saveOnZoom: true,
-  });
-
-  const currentEventSeries = useCurrentEventMarklineSeries({
-    event,
-    group,
-  });
-  const releaseSeries = useReleaseMarkLineSeries({group});
-  const flagSeries = useFlagSeries({
-    query: {
-      start: eventView.start,
-      end: eventView.end,
-      statsPeriod: eventView.statsPeriod,
-    },
-    event,
-  });
+      return createSeriesAndCount(unfilteredGroupStats['count()']);
+    }, [unfilteredGroupStats]);
+    const {series: unfilteredUserSeries} = useMemo(() => {
+      if (!unfilteredGroupStats?.['count_unique(user)']) {
+        return {series: []};
+      }
+      return createSeriesAndCount(unfilteredGroupStats['count_unique(user)']);
+    }, [unfilteredGroupStats]);
+    const userSeries = useMemo(() => {
+      if (!groupStats['count_unique(user)']) {
+        return [];
+      }
+
+      return createSeriesAndCount(groupStats['count_unique(user)']).series;
+    }, [groupStats]);
+
+    const chartZoomProps = useChartZoom({
+      saveOnZoom: true,
+    });
+
+    const currentEventSeries = useCurrentEventMarklineSeries({
+      event,
+      group,
+    });
+
+    const {releases = []} = useReleaseStats(
+      {
+        projects: eventView.project,
+        environments: eventView.environment,
+        datetime: {
+          start: eventView.start,
+          end: eventView.end,
+          period: eventView.statsPeriod,
+        },
+      },
+      {
+        staleTime: 0,
+      }
+    );
 
-  const series = useMemo((): BarChartSeries[] => {
-    const seriesData: BarChartSeries[] = [];
-    const translucentGray300 = Color(theme.gray300).alpha(0.5).string();
-    const lightGray300 = Color(theme.gray300).alpha(0.2).string();
+    const releaseSeries = useReleaseMarkLineSeries({
+      group,
+      releases: hasReleaseBubblesSeries && showReleasesAs !== 'line' ? [] : releases,
+    });
```

Comment:
I moved the `useReleaseStats` hook up into `<EventGraph>` so that 1) releases data is shared between the two release series (line/bubble) and 2) so that we can better control which display type to show (alternatively I could have introduced an enabled option to both hooks?).

Reply:
I think this is more transparent, so I don't see a fault in it. Good call on the sharing release data too 👍